lib/puppet/provider/foreman_resource/rest_v3.rb

```diff
@@ -95,6 +95,13 @@ def success?(response)
   end
 
   def error_message(response)
-    JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "unknown error (response #{response.code})"
+    case response.code
+    when '401'
+      '401 RESPONSE PLACEHOLDER ERROR MESSAGE'
+    when '503'
+      '503 RESPONSE PLACEHOLDER ERROR MESSAGE'
+    else
+      JSON.parse(response.body)['error']['full_messages'].join(' ') rescue "unknown error (response #{response.code})"
```

Comment:
It probably makes sense to always attempt to parse JSON and if see if it contains a full error message. This allows the Foreman application to give the most precise error in some scenarios. Alternatively, you can peek at the content type header to see if it actually is JSON.